File: Csharp/Hashtable/Q1.cs
﻿namespace Hashtable.Q1;

public class Solution1
{
    public int[] TwoSum(int[] nums, int target)
    {
        var hashtable = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++)
        {
            if (hashtable.ContainsKey(target - nums[i]))
            {
                return new int[] { hashtable[target - nums[i]], i };
            }
            hashtable[nums[i]] = i;
        }
        return new int[] { -1, -1 };
    }
}

File: Rust/src/hashtable/q1.rs
use std::collections::HashMap;

pub struct Solution1 {}

impl Solution1 {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut hashtable = HashMap::with_capacity(nums.len());
        for i in 0..Vec::len(&nums) {
            if let Some(key) = hashtable.get(&(target - nums[i])) {
                return vec![*key as i32, i as i32];
            }
            hashtable.insert(nums[i], i);
        }
        vec![]
    }
}
